packages/server/src/adapters/fastify/fastifyTRPCPlugin.ts

```diff
@@ -22,13 +22,15 @@ export function fastifyTRPCPlugin<TRouter extends AnyRouter>(
   opts: FastifyTRPCPluginOptions<TRouter>,
   done: (err?: Error) => void,
 ) {
-  fastify.addContentTypeParser(
-    'application/json',
-    { parseAs: 'string' },
-    function (_, body, _done) {
-      _done(null, body);
-    },
-  );
+  if (!fastify.hasContentTypeParser('application/json')) {
+    fastify.addContentTypeParser(
+      'application/json',
+      { parseAs: 'string' },
+      function (_, body, _done) {
+        _done(null, body);
+      },
+    );
+  }
```

Comment:
I'm not sure why this happens (I don't have much time to go into it), but my theory is that the content parser is explicitly defined by another plugin before the trpc adapter is initialized, and fastifiy doesn't allow you to change it even though you're in the scope of the trpc adapter. The fix I've found is to unregister it before adding it again for this scope.

```suggestion
  fastify.removeContentTypeParser('application/json');
  fastify.addContentTypeParser(
    'application/json',
    { parseAs: 'string' },
    function (_, body, _done) {
      _done(null, body);
    },
  );
```

Reply:
Probably related to https://github.com/nestjs/nest/commit/0444f9c89e61b5924cb4f9532303b9bb1d994787#diff-7601c6884e46e07a23aae0fea72039261b51395017f5ef74f58bb7a2519e5cb1R479 add 9 months ago

Reply:
I think this happens when nestjs puts the body parser on the Fastify root instance rather than on their adapter instance : https://github.com/nestjs/nest/commit/0444f9c89e61b5924cb4f9532303b9bb1d994787#diff-7601c6884e46e07a23aae0fea72039261b51395017f5ef74f58bb7a2519e5cb1R208

Reply:
https://github.com/nestjs/nest/blob/master/packages/platform-fastify/adapters/fastify-adapter.ts#L596

🤔